components/form/TaskForm.tsx

```diff
@@ -33,63 +38,51 @@ const validateTaskFields = (fields: TaskFields, projectId: number): Task => {
     throw Error("Invalid task form: missing required fields")
 }
 
-export function CreateTaskForm({
+function TaskForm({
+    task: taskOrNull,
+    onSubmit,
+    onCancel,
     project,
     projectId,
-    afterSubmit,
-    onCancel,
 }: TaskFormProps): JSX.Element {
-    const [taskFields, setTaskFields] = useState<TaskFields>({
-        project,
-        billable: true,
-    })
-    const { user } = useContext(UserContext)
-    const { postTask } = useUpdateTasks(user)
-
-    const [errorMessage, setErrorMessage] = useState<string>("")
-    const errorHandler = (error: Error) => setErrorMessage(`${error}`)
-
-    const onSubmit = async (event: React.MouseEvent) => {
+    const [taskFields, setTaskFields] = useState<TaskFields>(
+        taskOrNull || { project, billable: true }
+    )
+    const handleSubmit = (event: React.MouseEvent) => {
         event.preventDefault()
-        try {
-            const task = validateTaskFields(taskFields, projectId)
-            const newTask = await postTask(task, errorHandler)
-            return afterSubmit && afterSubmit(newTask)
-        } catch (error) {
-            errorHandler(error as Error)
-            return null
-        }
+        const updatedTask = validateTaskFields(taskFields, projectId)
+        onSubmit(updatedTask)
     }
-
     return (
-        <>
+        <form>
             <FormContainer>
-                <FormInputField
+                <Input
                     label={"Name"}
+                    value={taskFields?.name || ""}
                     placeholder={"Task name"}
                     isRequired={true}
                     isInvalid={!taskFields.name}
-                    formErrorMessage={"Task name cannot be empty."}
                     onChange={(event) =>
                         setTaskFields({
                             ...taskFields,
                             name: event.target.value,
                         })
                     }
-                    testId={"form-field-name"}
+                    data-testid={"form-field-name"}
                 />
-                <FormInputField
+                <Input
                     label={"Description"}
+                    value={taskFields?.description || ""}
                     placeholder={"Description"}
                     onChange={(event) =>
                         setTaskFields({
                             ...taskFields,
                             description: event.target.value,
                         })
                     }
-                    testId={"form-field-description"}
+                    data-testid={"form-field-description"}
                 />
-                <CheckBoxField
+                <Checkbox
```

Comment:
Oops, this looks broken. Let's NOT make this change, and retain `CheckBoxField` here. In other words, let's NOT change `CheckBoxField` to `Checkbox`.